src/main/java/fileio_use_case/application_business/timetable_specific_classes/TimetableGatewayInteractor.java

```diff
@@ -0,0 +1,105 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Flow;
+
+/** Interactor for TimetableGateway*/
+public class TimetableGatewayInteractor implements TimetableFileImportInputBoundary, Flow.Publisher<Object>{
+    ArrayList<Flow.Subscriber<Object>> receivers;
+    private final TimetableGateway timetableGateway;
+    /**
+     * Constructor
+     * @param timetableGateway - Pass in Timetable Gateway
+     */
+    public TimetableGatewayInteractor(TimetableGateway timetableGateway) {
+        this.timetableGateway = timetableGateway;
+        this.receivers = new ArrayList<>();
+    }
+    /**
+     * Given FileImportRequestModel, which holds a string of the JSON file path and
+     * given a session type, return a TimetableModel with specified session type
+     * from the JSON file.
+     * @param jsonData FileImportRequestModel, Session Type
+     * @return TimetableModel
+     */
+    public TimetableModel readFromFile(FileImportRequestModel jsonData, String sessionType) throws IOException, ParseException, java.text.ParseException, InvalidSectionsException {
+        String filePath = jsonData.getFilePath();
+        Timetable aTimetable = this.timetableGateway.readFromFile(filePath, sessionType);
+        for (Flow.Subscriber<Object> subscriber : receivers){
+            subscriber.onNext(aTimetable);
+        }
+        return timetableToTimetableModel(aTimetable);
+    }
+    /** Helper method for readFromFile */
+    private TimetableModel timetableToTimetableModel(Timetable aTimetable) {
+        List<CourseModel> allCourses = new ArrayList<>();
+        for (TimetableCourse course : aTimetable.getCourseList()) {
+            allCourses.add(timetableCourseToCourseModel(course));
+        }
+        return new TimetableModel(allCourses);
+    }
+    /** Helper method for timetableToTimetableModel */
+    private CourseModel timetableCourseToCourseModel(TimetableCourse course) {
+        List<SectionModel> allSections = new ArrayList<>();
+        for (Section section : course.getSections()) {
+            List<BlockModel> allBlocks = new ArrayList<>();
+            for (Block block : section.getBlocks()){
+                allBlocks.add(new BlockModel(block.getDay(), block.getStartTime(),
+                        block.getEndTime(), block.getRoom()));
+            }
+            allSections.add(new SectionModel(section.getCode(), section.getInstructorName(), allBlocks));
+        }
+        return new CourseModel(course.getTitle(),
+                allSections, course.getCourseSession(), course.getCourseCode(),
+                course.getBreadth());
+    }
+    /** Reads timetableRequestModel into a JSON file to be stored in
+     * src/main/saved_timetables/.
+     * @param timetableRequestModel - a request model for Timetable
+     */
+    public void timetableToFile(TimetableModel timetableRequestModel, String sessionType) throws InvalidSectionsException {
```

Comment:
This might be real;y difficult to test, but as a small sidenote for future: maybe we could think of a way to test the writing into JSON? (Atm I'm not aware of a way to do it or of a test for it being written. But maybe we could think about that!)